app/src/main/java/org/jellyfin/androidtv/startup/StartupActivity.java

```diff
@@ -70,6 +70,12 @@ protected void onCreate(Bundle savedInstanceState) {
             application.getPrefs().edit().putBoolean("pref_enable_info_panel", false).apply();
             application.getSystemPrefs().edit().putString("sys_pref_config_version", "4").apply();
         }
+        if (Integer.parseInt(application.getConfigVersion()) < 5) {
```

Comment:
Does it make sense to maintain a config version? I have been thinking it may be less overhead to just check if a specific preference exists and set the default value if it does not.

Reply:
I think it does make sense! Imagine a setting being renamed or (like in this case) a setting replacing another one. It would be annoying if the app auto-updates via the play store and suddenly your behavior changed because there is no migration. In this case external player users wouldn't use the external player anymore.

I do however think that the way this is done is ugly, in the future we might want to separate the preferences to a custom class which brings type safety (using enums instead of strings etc.) and handles migration a bit better.

This class could have getters & setters that need to be called and could use shared preferences as storage or maybe even sqlite.

Reply:
I've created a simple class for usage in my apps with these simple getters and setters and I've used it a lot. Might save a little bit of work https://github.com/Florianisme/SharedPreferences

Reply:
I would prefer the different settings being defined too, so no "getBoolean" or "getString" but "getPreferredPlayer" and "getConfigVersion".

Something like:
```kotlin
enum class PlayerType {
	AUTO,
	VLC,
	EXOPLAYER,
	EXTERNAL
}

class Preferences(
	private val context: Context
) {
	private val sharedPreferences by lazy {
		PreferenceManager.getDefaultSharedPreferences(context)
	}

	/* Preferences */
	var preferedPlayer by enumPreference("pref_video_player", PlayerType.AUTO)
	var configVersion by intPreference("sys_pref_config_version", 1)

	/** Storage functions */
	private fun intPreference(key: String, default: Int) = object : ReadWriteProperty<Preferences, Int> {
		override fun getValue(thisRef: Preferences, property: KProperty<*>): Int {
			return sharedPreferences.getInt(key, default)
		}

		override fun setValue(thisRef: Preferences, property: KProperty<*>, value: Int) {
			sharedPreferences.edit().putInt(key, value).apply()
		}
	}

	private inline fun <reified T: Enum<T>> enumPreference(key: String, default: T?) = object : ReadWriteProperty<Preferences, T?> {
		override fun getValue(thisRef: Preferences, property: KProperty<*>): T? {
			val stringValue = sharedPreferences.getString(key, null)

			return if (stringValue == null) default
			else T::class.java.enumConstants.find { it.name == stringValue }
		}

		override fun setValue(thisRef: Preferences, property: KProperty<*>, value: T?) {
			if (value == null) sharedPreferences.edit().remove(key).apply()
			else sharedPreferences.edit().putString(key, value.toString()).apply()
		}
	}
}
```

_I actually wanted to make a quick example but got carried away so this should probably work already lol_

Reply:
Now that I spend some effort I made an issue for it #250